opm/input/eclipse/Schedule/Action/ActionParser.cpp

```diff
@@ -17,324 +17,583 @@
+        std::string value{};
+    };
+
+    /// Index type for a sequence of tokens.
+    using TokenSize = std::vector<std::string>::size_type;
+
+    /// Type for representing the current token position.
+    ///
+    /// Signed type to allow '-1' to represent an invalid position.
+    using CurrentPos = std::make_signed_t<TokenSize>;
+
+    /// Sequence of concatenated condition string tokens of a single ACTIONX
+    /// condition block.
+    ///
+    /// Newline characters and other whitespace assume to be removed.
+    /// Constructor argument.
+    const std::vector<std::string>& tokens_;
+
+    /// Current token position (index into tokens_).
+    ///
+    /// Default value -1 represents an invalid position.
+    CurrentPos current_pos_ = -1;
+
+    /// Retrieve the current string token.
+    ///
+    /// Will throw an exception of type std::logic_error if the
+    /// parseIsComplete().
+    ///
+    /// \return tokens_[current_pos_].
+    [[nodiscard]] const std::string& currentToken() const;
+
+    /// Retrieve the current token type.
+    ///
+    /// \return Token type of currentToken().  Special 'end' token if the
+    /// parseIsComplete().
+    [[nodiscard]] Opm::Action::TokenType currentType() const;
+
+    /// Construct a parse tree node from current token.
+    [[nodiscard]] Node current() const;
+
+    /// Process completion predicate.
+    ///
+    /// \return Whether or not we have consumed all input tokens, i.e.,
+    /// whether or not the current_pos_ is within the valid range of tokens_
+    [[nodiscard]] bool parseIsComplete() const;
+
+    /// Finish current token and advance to next token position.
+    void advanceTokenPosition();
+
+    /// Finish current token and advance to next.
+    ///
+    /// Convenience function which calls advanceTokenPosition() and returns
+    /// current() at that position.
+    ///
+    /// \return Next token.
+    [[nodiscard]] Node next();
+
+    /// Parse an arithmetic comparison.
+    ///
+    /// Current token sequence is expected to be of the form
+    ///
+    ///    Some Expression <OP> Some Other Expression
+    ///
+    /// in which <OP> is an arithmetic comparison operator such as '<' or
+    /// '.EQ.'.
+    ///
+    /// This function consumes tokens from tokens_ and advances the current
+    /// token position until either the comparison expression is complete,
+    /// we encounter a failure condition, or exhaust the token sequence.
+    ///
+    /// \return Parse tree for an arithmetic comparison.  Will return an
+    /// invalid parse tree (TokenType::error) if any of the following
+    /// conditions hold
+    ///
+    ///  * The left or right expressions could not be successfully parsed.
+    ///  * The operator <OP> was not recognised as an arithmetic operator.
+    [[nodiscard]] Opm::Action::ASTNode parse_cmp();
+
+    /// Parse an arithmetic comparison operator.
+    ///
+    /// This function consumes a single tokens from tokens_ and advances the
```

Comment:
token

Reply:
> token

Thanks!  I've pushed an update to that effect.